packages/@aws-cdk/aws-codepipeline-actions/lib/stepfunctions/invoke-action.ts

```diff
@@ -0,0 +1,149 @@
+   * 
+   * @default - none
+   */
+  readonly stateMachineInput?: string | object;
+ 
+ 
+  /**
+   * Prefix (optinoal)
+   *
+   * By default, the action execution ID is used as the state machine execution name. 
+   * If a prefix is provided, it is prepended to the action execution ID with a hyphen and 
+   * together used as the state machine execution name.
+   */
+  readonly executionPrefixName?: string;
+}
+ 
+/**
+ * CodePipeline invoke Action that is provided by an AWS Step Function.
+ */
+export class StepFunctionsInvokeAction extends Action {
+  private readonly props: StepFunctionsInvokeActionProps;
+ 
+  constructor(props: StepFunctionsInvokeActionProps) {
+    super({
+      ...props,
+      resource: props.stateMachine,
+      category: codepipeline.ActionCategory.INVOKE,
+      provider: 'StepFunctions',
+      artifactBounds: {
+        minInputs: 0,
+        maxInputs: 1,
+        minOutputs: 0,
+        maxOutputs: 1,
+      },
+    });
+
+    //StateMachineInput is required when the StateMachineInputType is set as FilePath
+    if ( props.stateMachineInputType == StateMachineInputType.FILEPATH && (!props.stateMachineInput)) {
+      throw new Error('Input type FilePath was specified, but no filepath was provided');
+    }
+
+    //Input Artifact is required when the StateMachineInputType is set as FilePath
+    if ( props.stateMachineInputType == StateMachineInputType.FILEPATH && (!this.actionProperties.inputs)) {
+      throw new Error('Input type FilePath was specified, but no input artifact was provided');
+    }
+ 
+    this.props = props;
+  }
+ 
+  protected bound(_scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
+  codepipeline.ActionConfig {
+    // allow pipeline to invoke this step function
+    options.role.addToPolicy(new iam.PolicyStatement({
+      actions: ['states:StartExecution','states:DescribeStateMachine'],
+      resources: [this.props.stateMachine.stateMachineArn],
+    }));
+ 
+    // allow action executions to be inspected and invoked
+    options.role.addToPolicy(new iam.PolicyStatement({
+      actions: ['states:DescribeExecution'],
+      //TODO: find statemachinename from arn. ARN is part of interface IStateMachine which doesn't have the name property. Confirm with Matt
+      resources: [`arn:aws:states:*:*:execution:${this.props.stateMachine.stateMachineArn}:${this.props.executionPrefixName || ''}*`],
+    }));
+ 
+    this.props.stateMachine.grantStartExecution(options.role);
+ 
+    // allow the Role access to the Bucket, if there are any inputs/outputs
+    if ((this.actionProperties.inputs ?? []).length > 0) {
+      options.bucket.grantRead(options.role);
+    }
+    if ((this.actionProperties.outputs || []).length > 0) {
+      options.bucket.grantWrite(options.role);
+    }
+
+    return {
+      configuration: {
+        StateMachineArn: this.props.stateMachine.stateMachineArn,
+        Input: this.props.stateMachineInput,
```

Comment:
do you need to convert this from an object to a json-encoded string in the case where InputType is set to Literal?